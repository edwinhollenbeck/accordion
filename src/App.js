import React from 'react';

class App extends React.Component {
  state = {
    currentButtonIndex: 0
  };

  static defaultProps = {
    sections: []
  };

  handleButtonClick(i) {
    this.setState({currentButtonIndex: i});
  }

  renderButtons() {
    const {sections} = this.props
    return sections.map((section, i) => (
        <li key={i}>
          <button key={i} type='button' onClick={() => this.handleButtonClick(i)}>
            {section.title}
          </button>
        </li>
    ))
  }

  renderContent() {
    const currentButton = this.props.sections[this.state.currentButtonIndex]
    return (
      <p>{!!this.props.sections.length && currentButton.content}</p>
    )
  }

  render() {
    return (
      <ul>
        {this.renderButtons()}
        {this.renderContent()}
      </ul>
    );
  }
}

export default App;