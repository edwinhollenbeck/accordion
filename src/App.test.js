import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
})

it('renders without props', () => {
    const tree = renderer
        .create(<App />)
    expect(tree).toMatchSnapshot();
})

it('component opens a clicked section', () => {
    const wrapper = shallow(<App sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
})