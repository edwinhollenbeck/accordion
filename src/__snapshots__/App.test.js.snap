// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component opens a clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      key="0"
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      key="1"
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      key="2"
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
  <p>
    Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
  </p>
</ul>
`;

exports[`renders without props 1`] = `
<ul>
  <p />
</ul>
`;
